scripts/set_assignees.py

```diff
@@ -64,88 +64,91 @@ def process_pr(gh, maintainer_file, number):
     collab = list()
-    for a in ac:
-        collab += maintainer_file.areas[a].maintainers
-        collab += maintainer_file.areas[a].collaborators
+    for area in area_counter:
+        collab += maintainer_file.areas[area.name].maintainers
+        collab += maintainer_file.areas[area.name].collaborators
     collab = list(dict.fromkeys(collab))
     log(f"collab: {collab}")
 
-    sm = dict(sorted(maint.items(), key=lambda item: item[1], reverse=True))
+    _all_maintainers = dict(sorted(found_maintainers.items(), key=lambda item: item[1], reverse=True))
 
     log(f"Submitted by: {pr.user.login}")
-    log(f"candidate maintainers: {sm}")
-
-    maintainer = "None"
-    maintainers = list(sm.keys())
-
-    prop = 0
-    if maintainers:
-        maintainer = maintainers[0]
-
-        if len(ac) > 1 and list(ac.values())[0] == list(ac.values())[1]:
-            for aa in ac:
-                if 'Documentation' in aa:
-                    log("++ With multiple areas of same weight including docs, take something else other than Documentation as the maintainer")
-                    for a in all_areas:
-                        if (a.name == aa and
-                            a.maintainers and a.maintainers[0] == maintainer and
-                            len(maintainers) > 1):
-                            maintainer = maintainers[1]
-                elif 'Platform' in aa:
-                    log("++ Platform takes precedence over subsystem...")
-                    log(f"Set maintainer of area {aa}")
-                    for a in all_areas:
-                        if a.name == aa:
-                            if a.maintainers:
-                                maintainer = a.maintainers[0]
-                                break
-
-
-        # if the submitter is the same as the maintainer, check if we have
-        # multiple maintainers
-        if pr.user.login == maintainer:
-            log("Submitter is same as Assignee, trying to find another assignee...")
-            aff = list(ac.keys())[0]
-            for a in all_areas:
-                if a.name == aff:
-                    if len(a.maintainers) > 1:
-                        maintainer = a.maintainers[1]
-                    else:
-                        log(f"This area has only one maintainer, keeping assignee as {maintainer}")
-
-        prop = (maint[maintainer] / num_files) * 100
-        if prop < 20:
-            maintainer = "None"
-
-    log(f"Picked maintainer: {maintainer} ({prop:.2f}% ownership)")
-    log("+++++++++++++++++++++++++")
+    log(f"candidate maintainers: {_all_maintainers}")
+
+    maintainers = list(_all_maintainers.keys())
+    assignee = None
+
+    # we start with areas with most files changed and pick the maintainer from the first one.
+    # if the first area is an implementation, i.e. driver or platform, we
+    # continue searching for any other areas
+    for area in area_counter.items():
```

Comment:
You could do something like `for area, count in area_counter.items():`, then the code below would use variables with a meaningful name.